src/modules/expression-evaluator/src/evaluateExpression.ts

```diff
@@ -60,10 +50,27 @@ export default async function evaluateExpression(
     case '!=':
       return childrenResolved[0] != childrenResolved[1]
 
+    case 'CONCAT':
     case '+':
-      return childrenResolved.reduce((acc: number, child: number) => {
-        return acc + child
-      }, 0)
+      if (childrenResolved.length === 0) return childrenResolved
+
+      // Reduce based on "type" if specified
+      if (inputQuery?.type === 'string')
+        return childrenResolved.reduce((acc, child) => acc.concat(child), '')
+      if (inputQuery?.type === 'array')
+        return childrenResolved.reduce((acc, child) => acc.concat(child), [])
+
+      // Concatenate arrays/strings
+      if (childrenResolved.every((child) => typeof child === 'string' || Array.isArray(child)))
+        return childrenResolved.reduce((acc, child) => acc.concat(child))
+
+      // Merge objects
+      if (childrenResolved.every((child) => child instanceof Object && !Array.isArray(child))) {
+        return childrenResolved.reduce((acc, child) => ({ ...acc, ...child }), {})
+      }
+
+      // Or just try to add any other types
+      return childrenResolved.reduce((acc: number, child: number) => acc + child)
```

Comment:
should probably check if it's 'CONCAT' and do default string concat , otherwise  + ?

Reply:
I was trying to deprecate "CONCAT" completely, I only left it in as a fallback for backwards compatibility, so I'd rather they didn't behave differently.

The idea is, if you want to force concatenation, you set `type: "string"`  e.g.
```
{
operator: "+",
type: "string",
children: [1, 2, 3]
}
```
will return `"123"` (as opposed to `6`)

Reply:
ok, i was kind of referring to backwards compatibility, I am not sure if it's the case in our config, but concat logic is different now and it would add number and booleans instead of concating them. I guess it's not a biggy, i'll approve but would be good to check if we are concating numbers of some reason

Reply:
Yeah, I tried to cover a pretty wide range of possibilities in the test cases, and none of them have changed their output result, so we should be good. (We're mostly using stringSubstitution for string manipulation in the wild currently, and I just checked and all CONCAT usage in the templates are for simple string joins, no numbers)